flows/common/common.go

```diff
@@ -355,3 +381,8 @@ type stats struct {
 	// pkts indicates the total number of packets that have been sent.
 	Pkts *val
 }
+
+// OverridePortCreator sets a custom implementation of the port creator.
+func OverridePortCreator(pc PortCreator) {
```

Comment:
Ah, I see what you mean about this being a bit messy, but the only other option that I see is to have a global variable that is overridden - which seems the same level of exposure of internal implementation.

Reply:
the alternative i considered was changing  the Handle implementations to accept the PortCreator interface as a parameter, didn't seem worth the breaking change to do that

Reply:
Agreed, SGTM.